// pdf-parse/dist/cjs/geometry/Rectangle.cjs
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Rectangle = void 0;
const Line_js_1 = require('./Line.cjs');
const Point_js_1 = require('./Point.cjs');
const Shape_js_1 = require('./Shape.cjs');
class Rectangle extends Shape_js_1.Shape {
    from;
    width;
    height;
    constructor(from, width, height) {
        super();
        this.from = from;
        this.width = width;
        this.height = height;
    }
    get to() {
        return new Point_js_1.Point(this.from.x + this.width, this.from.y + this.height);
    }
    getLines() {
        const to = this.to;
        const lines = [
            new Line_js_1.Line(this.from, new Point_js_1.Point(to.x, this.from.y)),
            new Line_js_1.Line(this.from, new Point_js_1.Point(this.from.x, to.y)),
            new Line_js_1.Line(new Point_js_1.Point(to.x, this.from.y), to),
            new Line_js_1.Line(new Point_js_1.Point(this.from.x, to.y), to),
        ];
        return lines.filter((l) => l.valid);
    }
    transform(matrix) {
        const p1 = Shape_js_1.Shape.applyTransform([this.from.x, this.from.y], matrix);
        const p2 = Shape_js_1.Shape.applyTransform([this.from.x + this.width, this.from.y + this.height], matrix);
        const x = Math.min(p1[0], p2[0]);
        const y = Math.min(p1[1], p2[1]);
        const width = Math.abs(p1[0] - p2[0]);
        const height = Math.abs(p1[1] - p2[1]);
        this.from = new Point_js_1.Point(x, y);
        this.width = width;
        this.height = height;
        return this;
    }
}
exports.Rectangle = Rectangle;
